modules/local/mag_merge_samplesheet.nf

```diff
@@ -0,0 +1,28 @@
+process MAG_MERGE_SAMPLESHEET {
+
+    conda "conda-forge::sed=4.7"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/ubuntu:20.04' :
+        'nf-core/ubuntu:20.04' }"
+
+    input:
+    path ('samplesheets/*')
+
+    output:
+    path "*_samplesheet.csv", emit: samplesheet
+    path "versions.yml"   , emit: versions
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    head -n 1 `ls ./samplesheets/* | head -n 1` > ${prefix}_samplesheet.csv
```

Comment:
This worries me a little; because hacky awk.

Why not using the native `.collectFile()` operator: https://www.nextflow.io/docs/latest/operator.html#collectfile 

 would that be able to serve a similar purpose? Same applies to `mag_to_samplesheet` now I think about it... `collectFile` might serve both purposes in one, as long as the channel strucure is already correct.